pythran/pythonic/numpy/fft/irfft.hpp

```diff
@@ -0,0 +1,104 @@
+#ifndef PYTHONIC_NUMPY_FFT_IRFFT_HPP
+#define PYTHONIC_NUMPY_FFT_IRFFT_HPP
+
+#include "pythonic/include/numpy/fft/irfft.hpp"
+#include "pythonic/utils/functor.hpp"
+#include "pythonic/include/utils/array_helper.hpp"
+#include <array>
+#include <map>
+#include "pythonic/types/ndarray.hpp"
+#include "pythonic/__builtin__/None.hpp"
+#include "pythonic/__builtin__/ValueError.hpp"
+#include "pythonic/numpy/swapaxes.hpp"
+#include <string.h>
+#include <math.h>
+
+
+#include <stdio.h>
+#include "pythonic/numpy/fft/fftpack.c"
+
+
+PYTHONIC_NS_BEGIN
+
+namespace numpy
+{
+    namespace fft
+    {
+        
+        // Aux function
+        template<class T, size_t N>
+        types::ndarray<double, N> _irfft(types::ndarray<T, N> const &in_array, long NFFT, bool norm)
```

Comment:
this code is very similar to rfft, maybe you could aggregate stuff ?

Reply:
Sadly not! :( It's different enough that that would be ugly. For one things, rfft goes from real -> complex. Irfft goes from complex -> real. The sizes are also quite different for the two functions (rfft goes from N -> N/2+1, irfft goes from N -> 2(N-1) The only thing that's really shared are the twiddle factors.